sdcflows/transform.py

```diff
@@ -215,38 +215,33 @@ def to_displacements(self, ro_time, pe_dir):
             A NIfTI 1.0 object containing the distortion.
 
         """
-        from math import pi
-        from nibabel.affines import voxel_sizes, obliquity
-        from nibabel.orientations import io_orientation
-
-        # Generate warp field
-        data = self.shifts.get_fdata(dtype="float32").copy()
+        # Set polarity & scale VSM (voxel-shift-map) by readout time
+        vsm = self.shifts.get_fdata().copy()
         pe_axis = "ijk".index(pe_dir[0])
-        pe_sign = -1.0 if pe_dir.endswith("-") else 1.0
-        pe_size = self.shifts.header.get_zooms()[pe_axis]
-        data *= pe_sign * ro_time * pe_size
-
-        fieldshape = tuple(list(data.shape[:3]) + [3])
-
-        # Compose a vector field
-        field = np.zeros((data.size, 3), dtype="float32")
-        field[..., pe_axis] = data.reshape(-1)
-
-        # If coordinate system is oblique, project displacements through directions matrix
-        aff = self.shifts.affine
-        if obliquity(aff).max() * 180 / pi > 0.01:
-            dirmat = np.eye(4)
-            dirmat[:3, :3] = aff[:3, :3] / (
-                voxel_sizes(aff) * io_orientation(aff)[:, 1]
-            )
-            field = nb.affines.apply_affine(dirmat, field)
-
-        warpnii = nb.Nifti1Image(
-            field.reshape(fieldshape)[:, :, :, np.newaxis, :], aff, None
+        vsm *= -1.0 if pe_dir.endswith("-") else 1.0
+        vsm *= ro_time
+
+        # Shape of displacements field
+        fieldshape = tuple(list(vsm.shape[:3]) + [3])
```

Comment:
Given that we're adding a newaxis later, why not:

```suggestion
        # Note that ITK NIfTI fields are 5D (have an empty 4th dimension)
        fieldshape = tuple(list(vsm.shape[:3]) + [1, 3])
```